### PR TITLE
app and readme

### PR DESCRIPTION
**Pull request by napcs**
The Next.js app for the tutorial.

### CODE REVIEW

File: app/api/startBuy/route.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  try {
+    body = await req.json() as RequestBody;
+  } catch (error) {
+    return new Response("Invalid JSON body", { status: 400 });
+  }
+
+  const { itemId, transactionId } = body;
+
+  if (!itemId) {
+    return new Response("Must send the itemID to buy", { status: 405 });
```
Comment (angelazhou32) [inline app/api/startBuy/route.ts]: 405 is 'method not allowed' that doesn't super match with !itemId. How about we also use 400 insted?

File: app/page.tsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
+// @@@SNIPSTART typescript-next-oneclick-page-start
+'use client'
```
Comment (angelazhou32) [inline app/page.tsx:2]: Typo?

File: app/page.tsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+type ITEMSTATE = 'NEW' |  'ORDERED' | 'ERROR';
+// @@@SNIPEND
+
+// @@@SNIPSTART typescript-next-oneclick-page-product
+const Product: React.FC<ProductProps> = ({ product }) => {
+  const itemId = product.id;
+  const [state, setState] = React.useState<ITEMSTATE>('NEW');
+  const stateRef = React.useRef<ITEMSTATE>();
+  stateRef.current = state;
```
Comment (angelazhou32) [inline app/page.tsx]: Don't think we need lines 39-40

File: package.json [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    "build:temporal": "tsc --build ./temporal/tsconfig.json",
+    "start:worker": "nodemon ./temporal/lib/worker",
+    "start": "next start",
+    "lint": "eslint ."
+  },
+  "dependencies": {
+    "@temporalio/activity": "^1.9.3",
+    "@temporalio/client": "^1.9.3",
+    "@temporalio/worker": "^1.9.3",
+    "@temporalio/workflow": "^1.9.3",
```
Comment (angelazhou32) [inline package.json]: small nit but can use v1.10.0

File: temporal/src/activities.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+// @@@SNIPSTART typescript-next-oneclick-activities
+import { Context } from '@temporalio/activity';
+
+export async function purchase(id: string): Promise<string> {
+  console.log(`Purchased ${id}!`);
+  return Context.current().info.activityId;
```
Comment (angelazhou32) [inline temporal/src/activities.ts]: Can just do:

```
import { activityInfo } from '@temporalio/activity';

export async function purchase(id: string): Promise<string> {
      console.log(`Purchased ${id}!`);
       return activityInfo().activityId;
  }
  ```

File: temporal/src/workflows.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
+// @@@SNIPSTART typescript-next-oneclick-workflows
+import { proxyActivities, sleep } from '@temporalio/workflow';
+import type * as activities from './activities';
+
+const { purchase } = proxyActivities<typeof activities>({
+  startToCloseTimeout: '1 minute',
+});
+
+export async function OneClickBuy(id: string): Promise<string> {
```
Comment (angelazhou32) [inline temporal/src/workflows.ts]: camel case?

File: app/page.tsx [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+];
+
+type ITEMSTATE = 'NEW' |  'ORDERED' | 'ERROR';
+// @@@SNIPEND
+
+// @@@SNIPSTART typescript-next-oneclick-page-product
+const Product: React.FC<ProductProps> = ({ product }) => {
+  const itemId = product.id;
+  const [state, setState] = React.useState<ITEMSTATE>('NEW');
+  const stateRef = React.useRef<ITEMSTATE>();
```
Comment (angelazhou32) [inline app/page.tsx]: Slight nit but can just import useState, useRef if you wanted to, but nbd at all. IE:

```
import React, { useState, useRef  } from 'react';

const [state, setState] = useState<ITEMSTATE>('NEW');
```

File: app/api/startBuy/route.ts (hunk 1 above)
Comment (napcs) [reply app/api/startBuy/route.ts]:
```suggestion
    return new Response("Must send the itemID to buy", { status: 400 });
```

Comment (napcs) [reply app/api/startBuy/route.ts]: Agree.

File: app/page.tsx (hunk 2 above)
Comment (napcs) [reply app/page.tsx:2]: Nope, a directive for Next.js to make it a client component.

Comment (angelazhou32) [reply app/page.tsx:2]: Oops sorry, thanks!

**Review by angelazhou32 — APPROVED** [review]
Other than bumping to v1.10.0 which we have discussed, LGTM.

**Review by angelazhou32 — APPROVED** [review]
LGTM.

**Pull request merged**